bakery/codec_test.go

```diff
@@ -0,0 +1,61 @@
+package bakery
+
+import (
+	"math/rand"
+
+	gc "gopkg.in/check.v1"
+)
+
+type codecSuite struct{}
+
+var _ = gc.Suite(&codecSuite{})
+
+func (*codecSuite) TestJSONRoundTrip(c *gc.C) {
+	fpkp, err := GenerateKey()
+	c.Assert(err, gc.IsNil)
+	tpkp, err := GenerateKey()
+	c.Assert(err, gc.IsNil)
+
+	var rootKey [24]byte
+	n, err := rand.Read(rootKey[:])
+	c.Assert(err, gc.IsNil)
+	c.Assert(n, gc.Equals, 24)
+
+	cid, err := encodeJSONCaveatId(fpkp, caveatInfo{
+		PeerPublicKey: &tpkp.Public,
+		RootKey:       rootKey[:],
+		Predicate:     "is-authenticated-user",
+	})
+	c.Assert(err, gc.IsNil)
+
+	res, err := decodeCaveatId(tpkp, cid)
+	c.Assert(err, gc.IsNil)
+	c.Assert(res.PeerPublicKey, gc.IsNil)
+	c.Assert(res.RootKey, gc.DeepEquals, rootKey[:])
+	c.Assert(res.Predicate, gc.Equals, "is-authenticated-user")
+}
+
+func (*codecSuite) TestV0RoundTrip(c *gc.C) {
```

Comment:
I'd like to have more tests for some of the error paths here - caveat ids too short, bad decryption, encrypted data too show, different version numbers, etc.